### ISSUE TITLE
Is it possible to use a script file extension other than 'pg' with pg_prove?

### ISSUE THREAD

**Issue by polobo** [theory/pgtap#87]
pgtap v0.95.x
pg_prove 3.30

Doing either of the following seems to end up treating the SQL as Perl code.

`pg_prove --ext pgtap ~/tests/`
`pg_prove ~/tests/test*.pgtap`

Changing the test file extension to `pg` and just running `pg_prove ~/tests/` on the same file works.

**Comment by theory** [comment]
I don't understand. My test files always end in `.sql`, and when I run `pg_prove test/*.sql` it works perfectly. Can you send me an example package demonstrating the issue? Also, what versions of TAP-Parser-SourceHandler-pgTAP and App::Prove do you have installed?

**Comment by polobo** [comment]
**Environment**
```
Ubuntu: 14.04
Perl: 5.18.2
App::Prove: 3.35
TAP::Parser::SourceHandler::pgTAP: 3.30
PostgreSQL: 9.3
```
**The Script**
```
#!/usr/bin/env bash

export PGDATABASE='postgres'
psql -c "CREATE EXTENSION IF NOT EXISTS pgtap;"
function create_script() {
    FILE="{1}"
    echo '--Comment'
    echo 'BEGIN;'
    echo 'SELECT plan(1);'
    echo "SELECT ok(true);"
    echo 'SELECT * FROM finish();'
    echo 'ROLLBACK;'
}
if [ ! -d ./test-pg ]; then
    mkdir ./test-pg
fi
create_script > ./test-pg/test-source.pg
# All Good!
pg_prove ./test-pg

if [ ! -d ./test-pgtap ]; then
    mkdir ./test-pgtap
fi
create_script > ./test-pgtap/test-source.pgtap
# Thinks its Perl?
pg_prove --ext 'pgtap' ./test-pgtap
```
**My Output**
```
NOTICE:  extension "pgtap" already exists, skipping
CREATE EXTENSION
*****Running First pg_prove
test-pg/test-source.pg .. ok
All tests successful.
Files=1, Tests=1,  0 wallclock secs ( 0.02 usr  0.00 sys +  0.03 cusr  0.00 csys =  0.05 CPU)
Result: PASS
******Running Second pg_prove (w/ --ext)
Semicolon seems to be missing at test-pgtap/test-source.pgtap line 1.
Can't modify constant item in predecrement (--) at test-pgtap/test-source.pgtap line 2, near "Comment
BEGIN"
syntax error at test-pgtap/test-source.pgtap line 2, near "Comment
BEGIN"
Execution of test-pgtap/test-source.pgtap aborted due to compilation errors.
test-pgtap/test-source.pgtap .. 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 

Test Summary Report
-------------------
test-pgtap/test-source.pgtap (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=1, Tests=0,  1 wallclock secs ( 0.02 usr +  0.00 sys =  0.02 CPU)
Result: FAIL
```

**Comment by theory** [comment]
Ah, so I guess the issue is that `--ext` doesn't seem to work, yes?

**Comment by polobo** [comment]
At minimum.

Without research It seems like any extension except `pg` and `sql` (I know pg works and you claim sql works, but pgtap does not...) are treated in the implementation language while those two are possibly special-cased to be passed through to `psql`.

Specifying an explicit test file with a `pgtap` extension fails while the `pg` extension works.

`pg_prove --ext pgtap ./test-pg/test-source.pg`: ok; note I left --ext in place...

`pg_prove ./test-pgtap/test-source.pgtap`: nope

**Comment by theory** [comment]
Yeah, it likes `.pg`, `.sql`, and `.s`. `--ext` isn't getting passed through. Trying to figure out what to do about that.

**Comment by theory** [comment]
Okay, `--ext` is from App::Prove (`prove`), and just identifies what files to run, not the contents of those files. This allows you to run Perl and pgTAP tests in a single command:

    prove --source Perl \
          --ext .t --ext .pg \
          --source pgTAP --pgtap-option dbname=try \
                         --pgtap-option username=postgres \
                         --pgtap-option suffix=.pg

So what you want is --pgtap-suffix with `prove`. I think it might make sense to pass that option from `--ext` when you execute `pg_prove` instead of `prove`. The question is, since you can have multiple `--ext` options, but there is only one `pgtap-suffix` option, what to do if there is more than one `--ext`?

I think the answer is to support multiple peta-suffix options.

**Comment by theory** [comment]
I believe [this commit](https://github.com/theory/tap-parser-sourcehandler-pgtap/commit/e04b71b59b9c50d19917253d660b89538dcf9029) fixes the issue.

**Comment by theory** [comment]
I released [pg_prove 3.31](https://metacpan.org/release/TAP-Parser-SourceHandler-pgTAP) with this fix on Friday.
